File: app/src/main/java/artwork/java/ProgressBar.java
package artwork.java;

import android.content.Context;
import android.graphics.Canvas;
import android.graphics.Paint;
import android.util.AttributeSet;

public class ProgressBar extends androidx.appcompat.widget.AppCompatImageView {
    private int w, h, progress = 0;

    private Paint paint;
    public ProgressBar(Context context) {
        super(context);
        paint = new Paint(Paint.ANTI_ALIAS_FLAG);
    }
    public ProgressBar(Context context, AttributeSet attrs) {
        super(context, attrs);
        paint = new Paint(Paint.ANTI_ALIAS_FLAG);
    }
    @Override
    protected void onSizeChanged(int w, int h, int ow, int oh) {
        super.onSizeChanged(w, h, ow, oh);
        this.w = w;
        this.h = h;
    }
    public void setProgress(int i){
        this.progress = i;
        invalidate();
    }
    @Override
    protected void onDraw(Canvas canvas) {
        paint.setColor(0xff00ff00);
        canvas.drawRect(0,0, w*progress/100, h, paint);
        super.onDraw(canvas);
    }
}

File: app/src/main/java/artwork/kotlin/ProgressBar.kt
package artwork.kotlin

import android.content.Context
import android.graphics.Canvas
import android.graphics.Paint
import android.util.AttributeSet
import androidx.appcompat.widget.AppCompatImageView

class ProgressBar : AppCompatImageView {
    private var w = 0f
    private var h = 0f
    private var progress = 0f

    private var paint: Paint

    constructor(context: Context) : super(context) {
        paint = Paint(Paint.ANTI_ALIAS_FLAG)
    }

    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs) {
        paint = Paint(Paint.ANTI_ALIAS_FLAG)
    }

    override fun onSizeChanged(w: Int, h: Int, ow: Int, oh: Int) {
        super.onSizeChanged(w, h, ow, oh)
        this.w = w.toFloat()
        this.h = h.toFloat()
    }

    fun setProgress(i: Int) {
        this.progress = i.toFloat()
        invalidate()
    }

    override fun onDraw(canvas: Canvas) {
        paint.color = -0xff0100
        canvas.drawRect(0f, 0f, w * progress / 100, h, paint)
        super.onDraw(canvas)
    }
}
